website/mails/listeners.py

```diff
@@ -0,0 +1,57 @@
+"""Functions that listen for event signals and queue up emails.
+All triggered emails live here.
+"""
+from datetime import datetime
+
+from modularodm import Q
+
+from website import mails, settings
+from framework.auth import signals as auth_signals
+from website.project import signals as project_signals
+from website.conferences import signals as conference_signals
+
+
+@auth_signals.unconfirmed_user_created.connect
+def queue_no_addon_email(user):
+    """Queue an email for user who has not connected an addon after
+    `settings.NO_ADDON_WAIT_TIME` months of signing up for the OSF.
+    """
+    mails.queue_mail(
+        to_addr=user.username,
+        mail=mails.NO_ADDON,
+        send_at=datetime.utcnow() + settings.NO_ADDON_WAIT_TIME,
+        user=user,
+        fullname=user.fullname
+    )
+
+@project_signals.set_privacy_public.connect
+def queue_first_public_project_email(user, node):
+    """Queue and email after user has made their first
+    non-OSF4M project public.
+    """
+    sent_mail = mails.QueuedMail.find(Q('user', 'eq', user) & Q('sent_at', 'ne', None) &
+                                      Q('email_type', 'eq', 'new_public_project'))
```

Comment:
There's no saved constant for that, should I create the constants? or use something like ''' mails.NEW_PUBLIC_PROJECT['template'] '''? (<- I actually dont like that as it implies email type and template need to be the same, which they do not)